src/qengine/opencl.cpp

```diff
@@ -135,11 +139,12 @@ void QEngineOCL::clFinish(bool doHard)
     if (doHard) {
         queue.finish();
     } else {
-        for (unsigned int i = 0; i < (device_context->wait_events.size()); i++) {
-            device_context->wait_events[i].wait();
+        for (unsigned int i = 0; i < (device_context->wait_events->size()); i++) {
+            (*(device_context->wait_events.get()))[i].wait();
```

Comment:
Why the .get() and dereference here?

Reply:
If you can show me a more elegant way to effectively get to the `[]` operator, I'm open to it. (That's why.)